app/controllers/concerns/foreman/controller/normalize_controller_attributes.rb

```diff
@@ -0,0 +1,22 @@
+module Foreman::Controller::NormalizeControllerAttributes
+  extend ActiveSupport::Concern
+
+  private
+
+  def normalize_controller_attributes(attrs)
+    ctrls_and_vol = JSON.parse(attrs["controllers"]).
+      deep_transform_keys { |key| key.to_s.underscore }.
+      deep_symbolize_keys
+    volumes = {}
+    ctrls_and_vol[:volumes].each_with_index do |vol, index|
+      volumes[index.to_s] = vol
+    end
+    attrs["nvme_controllers"] = ctrls_and_vol[:controllers]&.select { |controller| controller[:type].include?("VirtualNVMEController") }
+    attrs["scsi_controllers"] = ctrls_and_vol[:controllers]&.select { |controller| !controller[:type].include?("VirtualNVMEController") }
```

Comment:
You can use [`#split`](https://apidock.com/rails/Array/split) method to divide the `controllers` part

Reply:
I have used `#partition` instead, wasn't able to understand the use case `#split`